flake.nix

```diff
@@ -24,372 +23,81 @@
+      packages = forAllSystems (system:
+        let
+          pkgs = nixpkgsFor.${system};
+          inherit (pkgs) lib callPackage stdenv fetchgit fetchFromGitHub;
+        in rec {
+          default = yosys;
 
-        abc-verifier = with final;
-          stdenv.mkDerivation rec {
-            pname = "abc-verifier";
-            version = "for-yosys-0.17";
+          ghdl = pkgs.ghdl;
 
+          abc-verifier = pkgs.abc-verifier.overrideAttrs (_: rec {
+            version = "yosys-0.17";
             src = fetchFromGitHub {
               owner = "yosyshq";
               repo = "abc";
-              rev = "09a7e6d";
+              rev = "09a7e6dac739133a927ae7064d319068ab927f90" # == version
+              ;
               hash = "sha256-+1UcYjK2mvhlTHl6lVCcj5q+1D8RUTquHaajSl5NuJg=";
             };
-
-            nativeBuildInputs = [ cmake ];
-            buildInputs = [ readline ];
-
-            installPhase = "mkdir -p $out/bin && mv abc $out/bin";
-
-            # needed by yosys
             passthru.rev = src.rev;
+          });
 
-            meta = with lib; {
-              description =
-                "A tool for squential logic synthesis and formal verification";
-              homepage = "https://people.eecs.berkeley.edu/~alanmi/abc";
-              license = licenses.mit;
-              maintainers = with maintainers; [ thoughtpolice ];
-              mainProgram = "abc";
-              platforms = platforms.unix;
-            };
-          };
-
-        yosys-ghdl = with final;
-          stdenv.mkDerivation {
-            pname = "yosys-ghdl";
-            # This is not the latest commit, but it's the latest that builds with current stable ghdl 2.0.0
-            version = "2022.01.11";
-
-            src = fetchFromGitHub {
-              owner = "ghdl";
-              repo = "ghdl-yosys-plugin";
-              rev = "c9b05e481423c55ffcbb856fd5296701f670808c";
-              sha256 = "sha256-tT2+DXUtbJIBzBUBcyG2sz+3G+dTkciLVIczcRPr0Jw=";
-            };
+          yosys-ghdl = pkgs.yosys-ghdl;
 
-            buildInputs = [ yosys readline zlib ghdl ];
-            nativeBuildInputs = [ pkg-config ];
-
-            doCheck = true;
-            installPhase = ''
-              mkdir -p $out/share/yosys/plugins
-              cp ghdl.so $out/share/yosys/plugins/ghdl.so
-            '';
-
-            meta = with lib; {
-              description = "GHDL plugin for Yosys";
-              homepage = "https://github.com/ghdl/ghdl-yosys-plugin";
-              license = licenses.isc;
-              platforms = platforms.all;
-              maintainers = with maintainers; [ thoughtpolice ];
-            };
-          };
-
-        yosys = with final;
-          stdenv.mkDerivation rec {
-            pname = "yosys";
+          # FIXME(jl): we're not sure why this yosys derivaiton is defined here,
+          # - is it the required version of yosys to use with nextpnr-xilinx?
```

Comment:
Calling out this comment, it's not explicit that `yosys` is a build or runtime dependency of `nextpnr-xilinx`, so we are wondering why we want it as part of the dev shell. Can you explain the workflow and how you'd imagine using these tools in the dev shell?

Reply:
nextpnr-xilinx needs yosys version 0.17, because later versions generate block ram primitives that nextpnr-xilinx does not yet understand.

Reply:
Thanks, this example is very useful!